### PR TITLE
Add suitable comments for GoDoc

### LINKED ISSUE #76: Add suitable comments for GoDoc
Add comments for the following,
- [x] Packages, a very brief description of its purpose
- [x] Exported functions and structs, detailed explanation on what it does and or represents.
- [x] Examples to use some important functions/structs

Assigning this to myself.

### PR DESCRIPTION
**Pull request by ayushjn20**
Resolves #76

### CODE REVIEW

**Comment by apoorvam** [comment]
@ayushjn20 Build is failing with this. Can you fix it so we can merge?

**Comment by ayushjn20** [comment]
Got it. I opened a PR from my own fork, that's why the _continuous-integration/travis-ci/pr_ is failing on snap login, but the _continuous-integration/travis-ci/push_ build is passing. 
I have pushed a branch on the original repo too, the build is passing. Kindly check [here](https://travis-ci.org/leopardslab/dunner/builds/551219105)
@agentmilindu @apoorvam I think we are good to go with this PR.

File: pkg/dunner/dunner.go [hunk 1]
```diff
@@ -129,7 +134,8 @@ func process(configs *config.Configs, s *docker.Step, wg *sync.WaitGroup, args [
 	}
 }
 
-func passArgs(s *docker.Step, args *[]string) error {
+// PassArgs replaces argument variables,of the form '`$d`', where d is a number, with dth argument.
+func PassArgs(s *docker.Step, args *[]string) error {
```
Comment (apoorvam) [inline pkg/dunner/dunner.go]: Do you want to expose these functions outside package? Its used only within the package.

Comment (ayushjn20) [reply pkg/dunner/dunner.go]: Yeah! If dunner is going to be used as a library, then the method that passes the arguments to the commands should be exposed.

**Pull request merged**